ember-bootstrap/addon/components/bs-progress/bar.ts

```diff
@@ -1,5 +1,22 @@
 import Component from '@glimmer/component';
 
+export interface ProgressBarSignature {
+  Args: {
+    minValue?: number;
+    maxValue?: number;
+    value?: number;
+    showLabel?: boolean;
+    striped?: boolean;
+    animate?: boolean;
+    roundDigits?: number;
+    type?: 'default' | 'success' | 'info' | 'warning' | 'danger' | string;
```

Comment:
This union type is effectively just `string`, so does not help much. Any typo like `defualt` would still be valid. Can't we list only the types Bootstrap supports ootb?

Reply:
We could change the `string` to `string & {}`[^1]. This would allow for users to use their own defined bar types while still enjoying the benefits of auto-complete, however I'm not sure how often that case is used. I'm fine with being more strict/correct

[^1]: https://stackoverflow.com/questions/74467392/autocomplete-in-typescript-of-literal-type-and-string

Reply:
Interesting, never heard of that!

But I still think being strict is probably better. Users that have this customized can still choose to not add our default types to their template registry, but have their of for a particular component they have extended, or extend our types. That is explicitly covered in the Glint docs [here](https://typed-ember.gitbook.io/glint/environments/ember/using-addons#using-glint-enabled-addons).